test/e2e/suite_test.go

```diff
@@ -0,0 +1,232 @@
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for middleware-monitoring to deploy
+	err = waitForInstallationStageCompletion(ctx.Client, retryInterval, monitoringStageTimeout, string(rhmiv1alpha1.MonitoringStage))
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for keycloak-operator to deploy
+	err = waitForProductDeployment(ctx.KubeClient, string(rhmiv1alpha1.ProductRHSSO), "keycloak-operator")
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for authentication phase to complete (10 minutes timeout)
+	err = waitForInstallationStageCompletion(ctx.Client, retryInterval, authenticationStageTimeout, string(rhmiv1alpha1.AuthenticationStage))
+	Expect(err).NotTo(HaveOccurred())
+
+	//Product Stage - verify operators deploy
+	products := map[string]string{
+		"3scale":               "3scale-operator",
+		"amq-online":           "enmasse-operator",
+		"codeready-workspaces": "codeready-operator",
+		"fuse":                 "syndesis-operator",
+		"user-sso":             "keycloak-operator",
+		"ups":                  "unifiedpush-operator",
+		"apicurito":            "apicurito-operator",
+	}
+
+	if installType == string(rhmiv1alpha1.InstallationTypeManagedApi) {
+		products = map[string]string{
+			"3scale":   "3scale-operator",
+			"user-sso": "keycloak-operator",
+		}
+	}
+
+	for product, deploymentName := range products {
+		err = waitForProductDeployment(ctx.KubeClient, product, deploymentName)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// wait for products phase to complete (30 minutes timeout)
+	err = waitForInstallationStageCompletion(ctx.Client, retryInterval, productsStageTimout, string(rhmiv1alpha1.ProductsStage))
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for solution-explorer operator to deploy
+	if installType == string(rhmiv1alpha1.InstallationTypeManaged) {
+		err = waitForInstallationStageCompletion(ctx.Client, retryInterval, solutionExplorerStageTimeout, string(rhmiv1alpha1.SolutionExplorerStage))
+		Expect(err).NotTo(HaveOccurred())
+	}
+	// +kubebuilder:scaffold:scheme
+
+	close(done)
+}, 5400)
+
+var _ = AfterEach(func() {
+	failed = failed || CurrentGinkgoTestDescription().Failed
+})
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+
+	ctx, err := common.NewTestingContext(cfg)
+
+	artifactsDir := os.Getenv(artifactsDirEnv)
+	if failed && artifactsDir != "" {
+		if _, err := os.Stat(artifactsDir); !os.IsNotExist(err) {
+			out := path.Join(artifactsDir, "rhmi.yaml")
+			By(fmt.Sprintf("Writing rhmi.yaml file to %s", out))
+			err = common.WriteRHMICRToFile(ctx.Client, out)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
+	err = testEnv.Stop()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+func waitForInstallationStageCompletion(k8sClient client.Client, retryInterval, timeout time.Duration, phase string) error {
+	err := wait.Poll(time.Second*5, time.Minute*30, func() (done bool, err error) {
```

Comment:
Doesn't look to use the `retryInterval` and `timeout time.Duration` parameters in the poll, so maybe either set these or remove as part of function if we're looking to have this the same for all stages :thinking: